### PR TITLE
add treemap graph

### PR DESCRIPTION
**Pull request by theiliad**

### CODE REVIEW

File: packages/core/demo/data/treemap.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+export const treemapData = [
+	{
+		name: "Oceania",
+		children: [
```
Comment (cal-smith) [inline packages/core/demo/data/treemap.ts:4]: Open question: are arbitrarily nested children allowed? I'm not super familiar with treemap variants, so I don't know if that would be desirable.

File: packages/core/src/configuration.ts [hunk 2]
```diff
@@ -364,6 +364,15 @@ const radarChart: RadarChartOptions = Tools.merge({}, chart, {
 	}
 } as RadarChartOptions);
 
+/**
+ * options specific to treemap charts
+ */
+const treemapChart: RadarChartOptions = Tools.merge({}, chart, {
```
Comment (cal-smith) [inline packages/core/src/configuration.ts:370]: Should this be `const treemapChart: TreeMapChartOptions`? Or is there sufficient overlap between tree and radar?

**Review by cal-smith — COMMENTED** [review]
One small comment, and one open question ... otherwise makes sense to me!

File: packages/core/demo/data/treemap.ts (hunk 1 above)
Comment (theiliad) [reply packages/core/demo/data/treemap.ts:4]: you could provide those children, however we haven't designed for that yet

**Comment by theiliad** [comment]
Designs already cleared with Jeannie, and colors with Sadek

**Comment by theiliad** [comment]
I've already cleared the designs

**Review by cal-smith — APPROVED** [review]

**Comment by sophiiae** [comment]
There's an `invariant violation` in React and it's blank in the Angular version.

**Review by cal-smith — APPROVED** [review]

**Comment by natashadecoste** [comment]
should we have a minimum tooltip width? Should it change widths **at all**? 
image
image
^ i think these can be the same widths

**Review by sophiiae — APPROVED** [review]

**Comment by natashadecoste** [comment]
Why would the user have to toggle **off** the label on the smaller "portions" rather than have the user toggle the label **on** for the bigger/more important slices? Seems theres always going to be more labels turned off than turned on (in most cases). If the user did nothing, theres more a chance the chart comes out looking messed up rather than if the default was not showing labels and the user **intentionally set those**. 
image
i think its rare that all the smaller children would be 2 character strings like this ^ but you can imagine with real names of places or companies there would be lots cut off

File: packages/core/src/model-treemap.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// Internal Imports
+import * as Configuration from "./configuration";
+// import { Tools } from "./tools";
+// import * as colorPalettes from "./services/colorPalettes";
+// import { Events, ScaleTypes } from "./interfaces";
+
+// // D3
+// import { map } from "d3-collection";
+// import { scaleOrdinal } from "d3-scale";
+// import { stack } from "d3-shape";
```
Comment (natashadecoste) [inline packages/core/src/model-treemap.ts:13]: we dont need yes?

File: packages/core/src/charts/treemap.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// Internal Imports
+import { TreemapChartModel } from "../model-treemap";
+import { Chart } from "../chart";
+import * as Configuration from "../configuration";
+import { ChartConfig, PieChartOptions } from "../interfaces/index";
```
Comment (natashadecoste) [inline packages/core/src/charts/treemap.ts:5]:
```suggestion
import { ChartConfig, TreemapChartOptions } from "../interfaces/index";
```

File: packages/core/src/charts/treemap.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	// the imports below are needed because of typescript bug (error TS4029)
+	Legend,
+	LayoutComponent,
+	Skeleton
+} from "../components/index";
+
+export class TreemapChart extends Chart {
+	model = new TreemapChartModel(this.services);
+
+	constructor(holder: Element, chartConfigs: ChartConfig<PieChartOptions>) {
```
Comment (natashadecoste) [inline packages/core/src/charts/treemap.ts:20]:
```suggestion
	constructor(holder: Element, chartConfigs: ChartConfig<TreemapChartOptions>) {
```

**Review by natashadecoste — CHANGES_REQUESTED** [review]

**Comment by theiliad** [comment]
> Why would the user have to toggle **off** the label on the smaller "portions" rather than have the user toggle the label **on** for the bigger/more important slices? Seems theres always going to be more labels turned off than turned on (in most cases). If the user did nothing, theres more a chance the chart comes out looking messed up rather than if the default was not showing labels and the user **intentionally set those**.
> image
> 
> i think its rare that all the smaller children would be 2 character strings like this ^ but you can imagine with real names of places or companies there would be lots cut off

if it's a demo issue we can merge as is and discuss that, then update in a later PR

**Review by natashadecoste — APPROVED** [review]

**Review by sophiiae — APPROVED** [review]

**Pull request merged**